src/Microsoft.DotNet.Build.Tasks/Targets/toolruntime.targets

```diff
@@ -24,9 +24,22 @@
     <ToolHostCmd Condition="'$(COMPLUS_InstallRoot)' != ''">(set COMPLUS_InstallRoot=) &amp; $(ToolHostCmd)</ToolHostCmd>
   </PropertyGroup>
 
+  <!-- If BUILDTOOLS_TOOLS_RUNTIME is set, then we will use that as the tool and test runtime, else we will restore it from NuGet. -->
+  <Target Name="UseBuildToolsRuntimeFromEnvVariable"
```

Comment:
> Currently this change will not enable that either

Yes, I provided feedback to that point.

> but we should be careful about mixing and matching these components. I agree we need to have a way to override but I'm not sure we want people to do it in a piece-wise fashion.

You will have a catch-22 for folks like Pat if you require that they bring an entire tool-runtime for their system but provide no way to build it.  Given that you will need a tool runtime to build managed code it seems like the best strategy is going to be restore the managed code using nuget and then fix up the native bits that are wrong for your system.

If you would rather provide a script that handles this (and just have folks override ToolRuntimePath), I'm fine with that approach, but it needs to be dead simple for me to produce a tool runtime on a machine where I don't have a functional version of CoreCLR to start with.